app/views/questionnaire.py

```diff
@@ -75,212 +73,122 @@ def save_questionnaire_store(response):
-    metadata = get_metadata(current_user)
-
-    if latest_location.block_id is 'summary':
-        aliases = SchemaHelper.get_aliases(g.schema_json)
-        schema_context = build_schema_context(metadata, aliases, answer_store)
-        rendered_schema_json = renderer.render(g.schema_json, **schema_context)
-        summary_context = build_summary_rendering_context(rendered_schema_json, answer_store, metadata)
-        return _build_template(current_location=latest_location, context=summary_context)
-
-    return redirect(latest_location.url(metadata))
-
-
-@questionnaire_blueprint.route('confirmation', methods=["GET"])
-@login_required
-def get_confirmation(eq_id, form_type, collection_id):  # pylint: disable=unused-argument
-    answer_store = get_answer_store(current_user)
-    path_finder = PathFinder(g.schema_json, answer_store, get_metadata(current_user))
-
-    latest_location = path_finder.get_latest_location(get_completed_blocks(current_user))
-
-    if latest_location.block_id == 'confirmation':
-        block = _render_schema(latest_location)
-        return _build_template(current_location=latest_location, context={"block": block})
-
-    metadata = get_metadata(current_user)
+        context = {'form': form, 'block': block}
+        return _build_template(current_location, context, template='questionnaire')
+    else:
+        questionnaire_store = get_questionnaire_store(current_user.user_id, current_user.user_ik)
+        update_questionnaire_store_with_answer_data(questionnaire_store, current_location, form.serialise(current_location))
 
-    return redirect(latest_location.url(metadata))
+        metadata = get_metadata(current_user)
+        path_finder = PathFinder(g.schema_json, get_answer_store(current_user), metadata)
+        next_location = path_finder.get_next_location(current_location=current_location)
+        return redirect(next_location.url(metadata))
 
 
 @questionnaire_blueprint.route('thank-you', methods=["GET"])
 @login_required
 def get_thank_you(eq_id, form_type, collection_id):  # pylint: disable=unused-argument
-    group_id = SchemaHelper.get_last_group_id(g.schema_json)
-    current_location = Location(group_id, group_instance=0, block_id='thank-you')
-    return _build_template(current_location=current_location, display_navigation=False, user_logout=True)
+    theme = g.schema_json.get('theme', None)
+    logger.debug("theme selected", theme=theme)
+    metadata = get_metadata(current_user)
+    metadata_context = build_metadata_context(metadata)
+    thank_you_template = render_theme_template(theme, template_name='thank-you.html',
+                                               meta=metadata_context,
+                                               schema_title=g.schema_json['title'],
+                                               legal_basis=g.schema_json['legal_basis'],
+                                               survey_id=g.schema_json['survey_id'])
```

Comment:
`analytics_ua_id=settings.EQ_UA_ID` is not being set in the new calls to `render_theme_template` so we'll be getting no analytics for these pages